cdap-proto/src/main/java/co/cask/cdap/proto/ProtoTrigger.java

```diff
@@ -192,6 +196,67 @@ public String toString() {
     }
   }
 
+  /**
+   * Represents a program status trigger for REST requests/responses
+   */
+  public static class ProgramStatusTrigger extends ProtoTrigger {
+    protected final ProgramId programId;
+    protected final Set<ProgramStatus> programStatus;
+
+    public ProgramStatusTrigger(ProgramId programId, Set<ProgramStatus> programStatus) {
```

Comment:
It represents the triggering program - good catch, I'll update the exception message below.